src/views/HomeView.vue

```diff
@@ -1,9 +1,158 @@
 <script setup lang="ts">
+import { ref, computed } from 'vue'
+import { useFirestore } from 'vuefire'
+import { collection, getDocs } from 'firebase/firestore'
+
+// FETCH ALL CATEGORIES AND SUBCATEGORIES
+const categories = ref<Array<{ name: string, iconURL: string, subCategories: Array<{ name: string, properties: [] }> }>>([]);
+
+const db = useFirestore()
+getDocs(collection(db, 'categories')).then((docs) => {
+  docs.forEach((doc) => {
+    console.log(doc.data());
+    categories.value.push({
+      name: doc.data().name,
+      iconURL: doc.data().icon_url,
```

Comment:
I had the same need to map between snake and camel cases with users and I solved it by changing the DB schema and logic to store the data in camelCase.

Snake_case is the accepted format for SQL databases, but camelCase will work just fine for nosql json docs.

Reply:
removed